src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlDateTimeMemberTranslator.cs

```diff
@@ -64,7 +72,8 @@ public NpgsqlDateTimeMemberTranslator(NpgsqlSqlExpressionFactory sqlExpressionFa
                     new[] { _sqlExpressionFactory.Constant("day"), instance! },
                     nullable: true,
                     argumentsPropagateNullability: TrueArrays[2],
-                    returnType),
+                    returnType,
+                    _typeMappingSource.FindMapping(typeof(DateTime))),
```

Comment:
This isn't quite correct: the PG `date_trunc` function returns the same PG type as it receives as a parameter:

```sql
SELECT pg_typeof(now()); -- timestamp with time zone
SELECT pg_typeof(date_trunc('day', now())); -- timestamp with time zone

SELECT pg_typeof(now() AT TIME ZONE 'utc'); -- timestamp without time zone
SELECT pg_typeof(date_trunc('day', now() AT TIME ZONE 'utc')); -- timestamp without time zone
```

Your proposed fix always sets the type mapping via CLR DateTime; that means you're setting the mapping as if `date_trunc` always returns `timestamp without time zone`. The correct fix would simply be to bubble up the type mapping of the provided instance:

```suggestion
                    instance!.TypeMapping),
```

Reply:
OK, I done the change